src/freenet/support/FileLoggerHook.java

```diff
@@ -988,17 +996,13 @@ else if(trace.length == 0)
 			else break;
 		}
 
-		try {
-			logString(sb.toString().getBytes(ENCODING));
-		} catch (UnsupportedEncodingException e1) {
-			throw new Error(e1);
-		}
+		logString(sb.toString().getBytes(ENCODING));
 	}
 
 	/** Memory allocation overhead (estimated through experimentation with bsh) */
 	private static final int LINE_OVERHEAD = 60;
 	
-	public void logString(byte[] b) throws UnsupportedEncodingException {
+	public void logString(byte[] b) {
```

Comment:
Theoretically, this change is not binary compatible…

Reply:
I restored this.
Yes, it is a checked exception, which removal may break callers.

This method should probably not be used directly from outside and there is formatted `log()` method instead.
What do you think about making this `@Deprecated` now and changing visibility to `private` in future.
Are there any known usages?

A new `logBytes()`  or  `appendBytes()` method without any exceptions may be also created and proposed as a new way how to use it,
but it is probably better to reduce its visibility.

What do you think?

Reply:
Yes, deprecate it for now, it doesn’t seem to be used anywhere else. And I don’t see any reason for adding another method that does this, the existence of this is absurd enough. 😄